project/Util.scala

```diff
@@ -0,0 +1,60 @@
+import sbt._
+import sbt.Keys._
+import java.nio.file.Files
+import java.util.regex.Pattern
+import scala.collection.JavaConverters._
+
+object Util {
+  lazy val copySharedResources = Def.task {
+//    Called when we have files under core, streaming that need to be copyied into services/js.
+//    ConsoleLogger().info("copying shared resources")
+//    val sharedMessages = "src/main/scala/org/apache/gearpump/shared/Messages.scala"
+//    IO.copyFile(core.base / sharedMessages, file(Build.servicesjs.baseDirectory) / sharedMessages)
+  }
+
+  def findDirs(currentDir: String): Array[File] = {
+    val dirs = IO.listFiles(file(currentDir),  DirectoryFilter)
+    dirs ++ dirs.flatMap(dir => {
+      val childDir = currentDir + "/" + dir.getName
+      findDirs(file(childDir).getPath)
+    })
+  }
+
+  def findFiles(dir: File, pattern: PatternFilter): Array[File] = {
+    val dirs = findDirs(dir.getPath)
+    IO.listFiles(dir, pattern) ++ dirs.flatMap(dir => {
+      IO.listFiles(dir, pattern)
+    })
+  }
+
+  def copyJSArtifactsToOutput: Unit = {
+    ConsoleLogger().info(s"copying JS artifacts ${Build.servicesjs.base.getPath}")
+    val in = file(s"${Build.servicesjs.base.getPath}/target/${Build.scalaVersionMajor}/gearpump-services-js-fastopt.js.map")
+    val out = file(s"${Build.distDashboardDirectory}/gearpump-services-js-fastopt.js.map")
+    val input = IO.read(in)
+    val data = input.replaceAll("../../src","src")
```

Comment:
I see here you repalce "../../src" with "src". This replacement is used for source mapping for debugging?

If that is the case, I suggest removing this part.
1. In the distributation package, there should be no .scala source files.